mlflow/tracing/types/wrapper.py

```diff
@@ -80,20 +72,22 @@ def status(self) -> SpanStatus:
     @property
     def inputs(self) -> Any:
         """The input values of the span."""
-        return self._inputs
+        input_serialized = self._span._attributes.get(SpanAttributeKey.INPUTS)
+        return json.loads(input_serialized) if input_serialized else None
```

Comment:
The property `span.inpus` and `span.outputs` returns the deserialized object for convenience. Instead of deserializing back, we can also save them as instance attribute like `self._inputs` and pass it to OTel's span when exporting. However, it makes discrepancy between OTel span and this wrapper which can cause unstable state. Most likely it is not common to read `inputs` / `outputs` the span during prediction, rather it is mainly done at evaluation time, then the cost of deserialization won't be too much of a concern.